lib/ruby_lsp/ruby_lsp_rails/server.rb

```diff
@@ -435,6 +437,29 @@ def clear_file_system_resolver_hooks
           ::ActionView::PathRegistry.file_system_resolver_hooks.clear
         end
       end
+
+      def collect_model_foreign_keys(model)
+        return [] unless model.connection.respond_to?(:supports_foreign_keys?) &&
+          model.connection.supports_foreign_keys?
+
+        model.connection.foreign_keys(model.table_name).map do |key_definition|
+          key_definition.options[:column]
+        end
+      rescue NotImplementedError
+        []
+      end
+
+      def collect_model_indexes(model)
+        model.connection.indexes(model.table_name).map do |index_definition|
+          {
+            name: index_definition.name,
+            columns: index_definition.columns,
+            unique: index_definition.unique,
+          }
+        end
+      rescue NotImplementedError
```

Comment:
Rather than having to `rescue` every time, let's cache that, e.g.: (not tested)

```ruby
def supports_indexing?(model)
  # We need to cache falsey values so can't use ||= for memoization
  return @db_supports_indexing if defined?(@db_supports_indexing)

  begin do
    model.connection.indexes(model.table_name)
    @db_supports_indexing = true
  rescue NotImplementedError
    @db_supports_indexing = false
  end
end
```